packages/adaptive_scaffold/lib/src/adaptive_layout.dart

```diff
@@ -0,0 +1,518 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+import 'package:flutter/widgets.dart';
+import 'breakpoints.dart';
+import 'slot_layout.dart';
+import 'slot_layout_config.dart';
+
+enum _SlotIds {
+  primaryNavigation,
+  secondaryNavigation,
+  topNavigation,
+  bottomNavigation,
+  body,
+  secondaryBody,
+}
+
+/// [AdaptiveLayout] is dedicated to simplifying the process of developing
+/// layouts for many different screen sizes. It separates the screen into
+/// several sections, also referred to as "slots". The Widget defines slots for
+/// the different placements of navigational elements and two slots for a body
+/// and secondaryBody. All of these slots take a [SlotLayout] Widget. This class
+/// works in junction with [SlotLayout] and [SlotLayoutConfig]. This class
+/// handles the placement of the slots on the screen and animations regarding
+/// their macromovements.
```

Comment:
```suggestion
/// Layout a typical app that adapt to different screens using predefined slots.
///
/// This widget separates the screen into predefined sections called "slots".
/// It layouts the app using the following kinds of slots (in order):
///
///  * [primaryNavigation], full width at the top.
///  * [secondaryNavigation], full width at the top, just below [primaryNavigation].
///  * (...)
///
/// Since slots must display differently under different screen conditions (such as different widths),
/// and each slot is defined with a [SlotLayout], which maps [Breakpoint]s to [SlotLayoutConfig],
/// where [SlotLayoutConfig] defines the content and transition.
///
/// [AdaptiveLayout] handles the placement of the slots on the screen and animations regarding
/// their macromovements.
///
/// For a preset of animations, breakpoints, see [AdaptiveScaffold], which provides a more 
/// friendly API with less customizability.
```

Reply:
Maybe link your example for adaptive layout here, although I'm not sure if dart docs parses it for plugins (maybe ask @/gspencergoog who wrote the tooling.)

Reply:
No, dartdoc doesn't use the snippet tool in packages.  Maybe it should someday, but right now it doesn't.  That's one drawback of not putting it in the main repo.